cmd/agent/gui/checks.go

```diff
@@ -185,56 +199,49 @@ func setCheckConfigFile(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Write new configs to custom checks directory
+	// Check if this file is in a subdir (file name is of the form `{check name}: {file name}`)
+	if i := strings.Index(fileName, ":"); i != -1 {
+		checkName := fileName[:i]
+		fileName = checkName + ".d/" + strings.TrimPrefix(fileName, checkName+": ")
+	}
+
+	// Attempt to write new configs to custom checks directory
 	path := config.Datadog.GetString("confd_path") + "/" + fileName
 	e = ioutil.WriteFile(path, data, 0644)
+
+	// If the write didn't work, try writing to the default checks directory
+	if e != nil && strings.Contains(e.Error(), "no such file or directory") {
+		path = filepath.Join(common.GetDistPath(), "conf.d") + "/" + fileName
+		e = ioutil.WriteFile(path, data, 0644)
```

Comment:
Configuration file might contain secret/password/token, we should set the rights to `0600`